docs/source/working_with_aiida/caching.rst

```diff
@@ -4,45 +4,92 @@
 Caching
 *******
 
-When working with AiiDA, you might sometimes re-run calculations which were already successfully executed. Because this can waste a lot of computational resources, you can enable AiiDA to **cache** calculations, which means that it will re-use existing calculations if a calculation with the same inputs is submitted again.
+Enabling caching
+----------------
 
-When a calculation is cached, a copy of the original calculation is created. This copy will keep the input links of the new calculation. The outputs of the original calculation are also copied, and linked to the new calculation. This allows for the new calculation to be a separate Node in the provenance graph and, critically, preserves the acyclicity of the graph.
+There are numerous reasons why you may need to re-run calculations you’ve already done before.
+Since AiiDA stores the full provenance of each calculation, it can *detect* whether a calculation has been run before and reuse its outputs without wasting computational resources.
+This is what we mean by **caching** in AiiDA.
 
-Caching is also implemented for Data nodes. This is not very useful in practice (yet), but is an easy way to show how the caching mechanism works:
+This feature is not enabled by default. In order to enable caching:
 
-.. ipython::
-    :verbatim:
+ 1. Place the following ``cache_config.yml`` file in your ``.aiida`` configuration folder
+    in order to enable caching for your AiiDA profile (here called ``aiida2``):
 
-    In [1]: from __future__ import print_function
+    .. code:: yaml
 
-    In [2]: from aiida.orm import Str
+        aiida2:
+          default: True
 
-    In [3]: n1 = Str('test string')
+ 2. Run ``verdi -p aiida2 rehash`` in order to compute hashes of all nodes in your existing database.
```

Comment:
I don't think step 2 is really necessary. Not even for older databases from before caching existed since there are migrations to compute the hash. If you want to keep it, maybe add a note that this is optional and may take a long time for big databases.

Reply:
I don't quite follow.
So, say caching is disabled, and then I enable it by putting the `.yml` file. 
Will all my nodes be automatically hashed the next time I do anything in AiiDA?
That would be strange, no?

Reply:
Hashing and caching are two different concepts. The hash of a node is _always_ computed and stored as an extra, regardless whether caching is enabled for that node type. That means that even when you do not have the caching config file yet, all nodes already should have their hash computed, so there is no need to call `verdi rehash`

Reply:
Ah, thanks for the clarification. 
I thought hashing was turned on using the same switch.